src/java.base/share/classes/java/lang/Class.java

```diff
@@ -1296,17 +1296,21 @@ private Class<?> elementType() {
      * {@code abstract} and {@code interface}; they should be decoded
      * using the methods of class {@code Modifier}.
      *
-     * <p> If the underlying class is an array class, then its
-     * {@code public}, {@code private} and {@code protected}
-     * modifiers are the same as those of its component type.  If this
-     * {@code Class} object represents a primitive type or void, its
-     * {@code public} modifier is always {@code true}, and its
-     * {@code protected} and {@code private} modifiers are always
-     * {@code false}. If this {@code Class} object represents an array class, a
-     * primitive type or void, then its {@code final} modifier is always
-     * {@code true} and its interface modifier is always
-     * {@code false}. The values of its other modifiers are not determined
-     * by this specification.
+     * <p> If the underlying class is an array class:
+     * <ul>
+     * <li> its {@code public}, {@code private} and {@code protected}
+     *      modifiers are the same as those of its component type
+     * <li> its {@code final} modifier is always
+     *      {@code true}
+     * <li> its interface modifier is always {@code false}, event when
+     *      the component type is an interface
```

Comment:
Should this mention that all array classes will also have their `abstract` modifier set?  The test cases seem to confirm that it's expected but we haven't stated it explicitly here.